Fix cross-compile

Original patch for DRI
http://git.buildroot.net/buildroot/commit/package/x11r7/xdriver_xf86-video-r128/xdriver_xf86-video-r128-cross-compile.patch?id=c0726391523da6a17ffa8e62681ffa0cc895da8e

Signed-off-by: Paulius Zaleckas <paulius.zaleckas@gmail.com>
(Added exa.h fix)
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

--- a/configure.ac	2008-06-23 10:39:28.000000000 -0400
+++ b/configure.ac	2008-06-23 10:40:46.000000000 -0400
@@ -85,14 +85,20 @@ sdkdir=$(pkg-config --variable=sdkdir xo
 AC_HEADER_STDC
 
 if test "$DRI" != no; then
-	AC_CHECK_FILE([${sdkdir}/dri.h],
+ 	if test "$cross_compiling" = "no" ; then
+		AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
-	AC_CHECK_FILE([${sdkdir}/sarea.h],
+		AC_CHECK_FILE([${sdkdir}/sarea.h],
                       [have_sarea_h="yes"], [have_sarea_h="no"])
-	AC_CHECK_FILE([${sdkdir}/dristruct.h],
+		AC_CHECK_FILE([${sdkdir}/dristruct.h],
                       [have_dristruct_h="yes"], [have_dristruct_h="no"])
-	AC_CHECK_FILE([${sdkdir}/damage.h],
+		AC_CHECK_FILE([${sdkdir}/damage.h],
                       [have_damage_h="yes"], [have_damage_h="no"])
+ 	else
+ 		have_dri_h="yes"
+ 		have_sarea_h="yes"
+ 		have_dristruct_h="yes"
+ 	fi
 fi
 
 AC_MSG_CHECKING([whether to include DRI support])
@@ -121,7 +127,12 @@
 AC_MSG_CHECKING([whether to enable EXA support])
 if test "x$EXA" = xyes; then
         AC_MSG_RESULT(yes)
-        AC_CHECK_FILE(${sdkdir}/exa.h, [have_exa_h="yes"], [have_exa_h="no"])
+	if test "$cross_compiling" = "no" ; then
+		AC_CHECK_FILE(${sdkdir}/exa.h,
+                      [have_exa_h="yes"], [have_exa_h="no"])
+	else
+		have_exa_h="yes"
+	fi
 else
         AC_MSG_RESULT(no)
 fi
